impala/hiveserver2.py

```diff
@@ -1214,54 +1253,62 @@ def update_has_result_set(self, state):
     def get_status(self):
         # pylint: disable=protected-access
         req = TGetOperationStatusReq(operationHandle=self.handle)
-        resp = self._rpc('GetOperationStatus', req)
+        # GetOperationStatus rpc is idempotent and so safe to retry.
+        resp = self._rpc('GetOperationStatus', req, True)
         self.update_has_result_set(resp)
         return TOperationState._VALUES_TO_NAMES[resp.operationState]
 
     def get_state(self):
         req = TGetOperationStatusReq(operationHandle=self.handle)
-        resp = self._rpc('GetOperationStatus', req)
+        # GetOperationStatus rpc is idempotent and so safe to retry.
+        resp = self._rpc('GetOperationStatus', req, True)
         self.update_has_result_set(resp)
         return resp
 
     def get_log(self, max_rows=1024, orientation=TFetchOrientation.FETCH_NEXT):
         try:
             req = TGetLogReq(operationHandle=self.handle)
-            log = self._rpc('GetLog', req).log
+            # GetLog rpc is idempotent and so safe to retry.
+            log = self._rpc('GetLog', req, True).log
         except TApplicationException as e: # raised if Hive is used
             if not e.type == TApplicationException.UNKNOWN_METHOD:
                 raise
             req = TFetchResultsReq(operationHandle=self.handle,
                                    orientation=orientation,
                                    maxRows=max_rows,
                                    fetchType=1)
-            resp = self._rpc('FetchResults', req)
+            resp = self._rpc('FetchResults', req, False)
             schema = [('Log', 'STRING', None, None, None, None, None)]
             log = self._wrap_results(resp.results, schema, convert_types=True)
             log = '\n'.join(l[0] for l in log)
         return log
 
     def cancel(self):
         req = TCancelOperationReq(operationHandle=self.handle)
-        return self._rpc('CancelOperation', req)
+        # CancelOperation rpc is idempotent and so safe to retry.
+        return self._rpc('CancelOperation', req, True)
 
     def close(self):
         req = TCloseOperationReq(operationHandle=self.handle)
-        return self._rpc('CloseOperation', req)
+        # CloseImpalaOperation rpc is not idempotent for dml and we're not sure
```

Comment:
nit: The comment should be about "CloseOperation".

Reply:
done